### PR TITLE
Adding adstex

### PR DESCRIPTION
**Pull request by yymao**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble be please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/adstex/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - ads >=0.12.3
+    - bibtexparser >=0.6.2
+    - future >=0.12.0  # [py<30]
```
Comment (dopplershift) [inline recipes/adstex/meta.yaml:26]: Do you really need Python 2.7 support? Right now conda-forge isn't even building Python 2.7 packages by default.

Comment (yymao) [reply recipes/adstex/meta.yaml:26]: No. Python 3+ is fine. I actually did that in 48fdb2a3 but the linter said that "Non noarch packages should have python requirement without any version constraints." So I then removed the Python version requirement.  

Can you suggest what may be a good way to edit the recipe? Thank you!.

Comment (dopplershift) [reply recipes/adstex/meta.yaml:26]: Well for this line, you should be able to completely eliminate the dependency on `future`.

File: recipes/adstex/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/adstex-{{ version }}.tar.gz
+  sha256: 9a192841625e4f79480bf430c5a9631060f64dee4ece018f4a992cfe7d605e49
+
+build:
+  number: 0
```
Comment (dopplershift) [inline recipes/adstex/meta.yaml:14]: Add `noarch: python`

File: recipes/adstex/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  entry_points:
+    - adstex=adstex:main
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - pip
+    - python
```
Comment (dopplershift) [inline recipes/adstex/meta.yaml:22]: Can become `python >=3.6`

File: recipes/adstex/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  run:
+    - ads >=0.12.3
+    - bibtexparser >=0.6.2
+    - future >=0.12.0  # [py<30]
+    - python
```
Comment (dopplershift) [inline recipes/adstex/meta.yaml:27]: Also `python >=3.6`

File: recipes/adstex/meta.yaml (hunk 1 above)
Comment (yymao) [reply recipes/adstex/meta.yaml]: Thank you for the tips! I've implemented your suggestions.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**